schema/ksl-schema.kdl

```diff
@@ -0,0 +1,925 @@
+                                MUST NOT fail.
+                                """ {
+                                repeatable
+                                arg about="ID reference to another node" {
+                                    type string
+                                    optional
+                                }
+                                prop path about="KPath reference to another node." {
+                                    type string
+                                    format kpath
+                                }
+                                prop base about="""
+                                    Base schema
+
+                                    The schema to resolve references against. If not
+                                    provided, the base schema SHALL be the one
+                                    defined in `metadata > id` for the current
+                                    schema.
+
+                                    Relative schema references SHALL be resolved
+                                    against `metadata > id`.
+                                    """ {
+                                    type string
+                                    format url-reference irl-reference
+                                }
+                                children {
+                                    node id about="ID reference to another node." {
+                                        repeatable
+                                        arg {
+                                            type string
+                                        }
+
+                                    }
+                                    node path about="KPath reference to another node." {
+                                        repeatable
+                                        arg {
+                                            type string
+                                            format kpath
+                                        }
+                                    }
+                                }
+                            }
+                            // TODO: be more precise and clear about merge + override behavior.
+                            node default about="""
+                                Definition default
+
+                                Any children of this node will be merged into the node
+                                definition itself, as if written directly into the node.
+                                Any conflicting definitions will be overriden by
+                                definitions within the rest of the node.
+                                """ {
+                                children {
+                                    allow-others
+                                }
+                            }
+                            node undefine about="Undefine a node with this name" {
```

Comment:
If we get rid of default stuff, can we remove this too?

Reply:
No, because it's still valuable for reversing definitions when using `ref`. ("I want this node definition, except for this one child")